driver_test.go

```diff
@@ -862,6 +865,108 @@ func testString(t *testing.T, json bool) {
 	})
 }
 
+/** TESTING TYPES **/
+// testUUID is a wrapper around UUID for unit testing purposes and should not be used in production
+type testUUID struct {
+	UUID
+}
+
+func newTestUUID() testUUID {
+	return testUUID{NewUUID()}
+}
+
+func parseTestUUID(str string) testUUID {
+	return testUUID{ParseUUID(str)}
+}
+
+// Scan implements sql.Scanner so UUIDs can be read from databases transparently.
+// Currently, database types that map to string and []byte are supported. Please
+// consult database-specific driver documentation for matching types.
+func (uuid *testUUID) Scan(src interface{}) error {
+	switch src := src.(type) {
+	case nil:
+		return nil
+
+	case string:
+		// if an empty UUID comes from a table, we return a null UUID
+		if src == "" {
+			return nil
+		}
+
+		// see Parse for required string format
+		u := ParseUUID(src)
+
+		*uuid = testUUID{u}
+
+	case []byte:
+		// if an empty UUID comes from a table, we return a null UUID
+		if len(src) == 0 {
+			return nil
+		}
+
+		// assumes a simple slice of bytes if 16 bytes
+		// otherwise attempts to parse
+		if len(src) != 16 {
+			return uuid.Scan(string(src))
+		}
+		copy((uuid.UUID)[:], src)
+
+	default:
+		return fmt.Errorf("Scan: unable to scan type %T into UUID", src)
+	}
+
+	return nil
+}
+
+// Value implements sql.Valuer so that UUIDs can be written to databases
+// transparently. Currently, UUIDs map to strings. Please consult
+// database-specific driver documentation for matching types.
+func (uuid testUUID) Value() (driver.Value, error) {
+	return uuid.String(), nil
+}
+
+func TestUUID(t *testing.T) {
+	testUuid(t, false)
```

Comment:
How about testing with `true`? It should work with arrow response as well. So maybe `TestUUIDInJson` and `TestUUIDInArrow`?